docs/sql-reference/10-sql-commands/00-ddl/07-virtual-column/index.md

```diff
@@ -0,0 +1,98 @@
+---
+title: VIRTUAL COLUMN
+---
+import IndexOverviewList from '@site/src/components/IndexOverviewList';
+import EEFeature from '@site/src/components/EEFeature';
+
+<EEFeature featureName='VIRTUAL COLUMN'/>
+
+A virtual column is a construct formed by extracting nested fields within [Variant](../../../00-sql-reference/10-data-types/43-data-type-variant.md) data and storing that data in separate storage files. Virtual columns offer enhanced efficiency in querying Variant data, with the following key benefits:
+
+- **Accelerated Query Processing**: Virtual columns streamline the querying process by eliminating the need to traverse the entire nested structure to locate the desired data. Direct data retrieval from virtual columns parallels the process of accessing regular columns, resulting in a significant acceleration of query execution.
+
+- **Reduced Memory Usage**: Variant data often includes numerous internal fields, and reading all of them can lead to substantial memory consumption. By transitioning to reading virtual columns, there is a notable reduction in memory usage, mitigating the risk of potential memory overflows.
+
+![Alt text](../../../../public/img/sql/virtual-column.png)
+
+### Managing Virtual Columns
+
+Databend provides the following commands to manage virtual columns:
+
+<IndexOverviewList />
+
+### Usage Example
+
+This example demonstrates the practical use of virtual columns and their impact on query execution:
+
+```sql
+-- Create a table named 'test' with columns 'id' and 'val' of type Variant.
+CREATE TABLE test(id int, val variant);
```

Comment:
I am not creating any virtual columns, but the explain show it has
<img width="841" alt="image" src="https://github.com/datafuselabs/databend-docs/assets/172204/d9629345-bd59-446f-9fd2-ca6b429969ed">

What's the differences to create virtual column? @b41sh

Reply:
The virtual columns explain show need improved by @b41sh later.
If we don't create virtual columns, the explain will not show them in the push_downs.(Actual now this is a fake push down, indeed there is no virtual columns in the storage).

@soyeric128 we can wait this PR after the new fix merge.

Reply:
has fixed in PR https://github.com/datafuselabs/databend/pull/14147